lib/core/test/unit/Cardano/Pool/MetricsSpec.hs

```diff
@@ -62,6 +69,28 @@ spec = do
     describe "calculatePerformances" $ do
         it "performances are always between 0 and 1"
             $ property prop_performancesBounded01
+        it "50% stake, producing 4/8 blocks => performance=1" $ do
+           let p = calculatePerformance 8 stake (productions 4)
+           Map.lookup pool p `shouldBe` (Just 1)
+        it "50% stake, producing 2/8 blocks => performance=0.5" $ do
+           let p = calculatePerformance 8 stake (productions 2)
+           Map.lookup pool p `shouldBe` (Just 0.5)
+        it "50% stake, producing 0/8 blocks => performance=0" $ do
+           let p = calculatePerformance 8 stake (productions 0)
+           Map.lookup pool p `shouldBe` (Just 0)
+  where
+    pool = PoolId "athena"
```

Comment:
I should factor this out to a separate helper of type `Spec`.

Ironically, when trying to add a "simple unit" test I ended up with something that would work very well as a property…

Reply:
> I ended up with something that would work very well as a property…

This is actually often the path ;). First write the simple stuff and then, see if there's any abstraction that seems to shine out of it. Trying to go for the abstraction first is often much harder and doesn't necessarily give better results.

Reply:
I should have learned by now 😶